File: test_samples/sample.go
package main

import "fmt"

func Add(a int, b int) int {
    return a + b
}

func Multiply(x float64, y float64) float64 {
    return x * y
}

func IsEven(n int) bool {
    return n%2 == 0
}

func main() {
    fmt.Println("Testing Go functions")
    fmt.Printf("Add(2, 3) = %d\n", Add(2, 3))
    fmt.Printf("Multiply(2.5, 4.0) = %f\n", Multiply(2.5, 4.0))
    fmt.Printf("IsEven(4) = %t\n", IsEven(4))
}

File: examples/sample.rs
pub fn add(a: i32, b: i32) -> i32 {
    a + b
}

pub fn multiply(x: f64, y: f64) -> f64 {
    x * y
}

pub fn is_even(n: u32) -> bool {
    n % 2 == 0
}

fn main() {
    println!("Sample functions for testing");
    println!("add(2, 3) = {}", add(2, 3));
    println!("multiply(2.5, 4.0) = {}", multiply(2.5, 4.0));
    println!("is_even(4) = {}", is_even(4));
}